### PR TITLE
[FIX] web: action_service: alwyas rethrow whatever error

### PR DESCRIPTION
**Pull request by kebeclibre**
on a legacy form view, do an action in target new
The view in target new should, during its onchange, trigger a ValidationError

Before this commit, the dialog_container was stuck in an error loop.
This was because the received error was never an instance of Error, rather it was
a legacy error event. So it was never transmitted upstream were the ErrorHandler could remove
the dialog in error from is dialog array.

After this commit, the ValidationError mesage is displayed.

Before this commi

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by kebeclibre** [comment]
robodoo rebase-ff

File: addons/web/static/src/core/debug/debug_menu.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
                     <t t-if="!element.hide">
                         <t t-if="element.type == 'item'">
                             <DropdownItem payload="{ callback: element.callback }">
-                                <a t-if="element.href" t-att-href="element.href" t-esc="element.description"/>
+                                <a t-if="element.href" t-att-href="element.href" t-esc="element.description" t-on-click.prevent=""/>
```
Comment (brboi) [inline addons/web/static/src/core/debug/debug_menu.xml:15]: Same fix must be applied for the user menu: support and documentation links are opened twice too.

File: addons/web/static/src/webclient/actions/action_service.js [hunk 2]
```diff
@@ -591,11 +599,14 @@ function makeActionManager(env) {
...
         const closingProm = _executeCloseAction();
-        env.bus.trigger("ACTION_MANAGER:UPDATE", {
+
+        const containerInfo = {
             id: ++id,
             Component: ControllerComponent,
             componentProps: controller.props,
-        });
+        };
+        controllerToInfo.set(controller, containerInfo);
```
Comment (kebeclibre) [inline addons/web/static/src/webclient/actions/action_service.js:609]: controller.__rendering_info__ =

File: addons/web/static/src/webclient/actions/action_service.js [hunk 3]
```diff
@@ -467,11 +468,18 @@ function makeActionManager(env) {
...
+                if (action.target === "new") {
+                    // get the dialog service to close the dialog.
                     throw error;
+                } else {
+                    const lastCt = controllerStack[controllerStack.length - 1];
+                    let info = {};
+                    if (lastCt) {
+                        info = controllerToInfo.get(lastCt);
+                    }
+                    env.bus.trigger("ACTION_MANAGER:UPDATE", info);
```
Comment (ged-odoo) [inline addons/web/static/src/webclient/actions/action_service.js:489]: write on controller the info, and remove weakmap

File: addons/web/static/tests/webclient/actions/load_state_tests.js [hunk 4]
```diff
@@ -927,4 +926,44 @@ QUnit.module("ActionManager", (hooks) => {
             );
         }
     );
+
+    QUnit.test("initial action crashes", async (assert) => {
+        assert.expect(7);
```
Comment (kebeclibre) [inline addons/web/static/tests/webclient/actions/load_state_tests.js:931]: recheck that flow in enterprise (fa-square should be displayed)

File: addons/base_automation/static/tests/base_automation_error_dialog.js [hunk 5]
```diff
@@ -36,14 +37,14 @@ QUnit.module("base_automation", {}, function () {
             // Both of these are unused but required for the error service to call error handlers
             serviceRegistry.add("notification", notificationService);
             serviceRegistry.add("rpc", makeFakeRPCService());
-            const windowAddEventListener = window.addEventListener;
-            window.addEventListener = (type, cb) => {
+            const windowAddEventListener = browser.addEventListener;
```
Comment (kebeclibre) [inline addons/base_automation/static/tests/base_automation_error_dialog.js:40]: split into another commit

**Comment by ged-odoo** [comment]
robodoo r+
